src/libmongoc/src/mongoc/mongoc-collection.c

```diff
@@ -3503,3 +3508,35 @@ mongoc_collection_watch (const mongoc_collection_t *coll,
 {
    return _mongoc_change_stream_new_from_collection (coll, pipeline, opts);
 }
+
+bool
+mongoc_collection_set_timeout_ms (mongoc_collection_t *coll,
+                                  int64_t timeout_ms,
+                                  bson_error_t *error)
+{
+   BSON_ASSERT_PARAM (coll);
+
+   if (timeout_ms < -1) {
```

Comment:
I think this can error if `timeout_ms < 0`.

According to the spec, users should not be able to set `timeoutMS` to unset after it has been set:

> Drivers MUST error if a negative value is specified. This value MUST be configurable at the level of a MongoClient, MongoDatabase, MongoCollection, or of a single operation. However, if the option is specified at any level, it cannot be later changed to unset.

Reply:
Oh good point!  Will change.